nemo_skills/pipeline/utils.py

```diff
@@ -1164,15 +1164,21 @@ def add_task(
         if isinstance(tunnel, run.SSHTunnel) and reuse_code:
             reuse_code_exp = reuse_code_exp or REUSE_CODE_EXP.get(tunnel_hash(tunnel))
             if reuse_code_exp is not None:
-                if isinstance(reuse_code_exp, run.Experiment):
-                    LOG.info("Reusing code from experiment %s", reuse_code_exp._title)
-                    reuse_dir = reuse_code_exp.tunnels[tunnel.key].packaging_jobs['nemo-run'].dst_path
+                reuse_code_exp = (
+                    reuse_code_exp
+                    if isinstance(reuse_code_exp, run.Experiment)
+                    else run.Experiment.from_title(reuse_code_exp)
```

Comment:
Does it work without context manager? This code is running from inside another `with run.Experiment`, so I was getting errors previously when using it like that

Reply:
I have now made the SlurmScheduler independent of context manager, but I will test the use case by providing a str as `reuse_code_exp`

Reply:
thanks, let me know if it works and we can merge after that

Reply:
Tested with both `title` and `id` strs i.e. (`exp._title` and `exp._id`) with https://github.com/NVIDIA/NeMo-Skills/pull/408/commits/5e1b991f1e96e24bc4c4d6ac505b5af09ad5c687